cmd/cosign/cli/public_key.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright The Sigstore Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/kms"
+
+	"github.com/peterbourgon/ff/v3/ffcli"
+)
+
+func PublicKey() *ffcli.Command {
+	var (
+		flagset = flag.NewFlagSet("cosign public-key", flag.ExitOnError)
+		key     = flagset.String("key", "", "path to the private key")
+		kmsVal  = flagset.String("kms", "", "sign via a private key stored in a KMS")
+	)
+
+	return &ffcli.Command{
+		Name:       "public-key",
+		ShortUsage: "cosign public key gets a public key from the key-pair [-kms KMSPATH]",
```

Comment:
nit: "cosign public-key" instead of "cosign public key" might be easier to read

Reply:
Yup! I always have a fear that I typo'd these and wrote "pubic". It's bound to happen.